apps/shared/avifpng.c

```diff
@@ -368,11 +384,33 @@ avifBool avifPNGRead(const char * inputFilename,
     return readResult;
 }
 
+//------------------------------------------------------------------------------
+// Writing
+
+// Does the opposite of avifCopyRawProfile(): writes a payload as a raw profile string.
+static avifBool avifBytesToRawProfile(const avifRWData * bytes, const char * profileName, avifRWData * profile)
+{
+    if (bytes->size > 99999999) {
+        return AVIF_FALSE;
+    }
+    size_t position = 1 + strlen(profileName) + 1 + 8 + 1;
+    const size_t profileLength = position + bytes->size * 2 + 1;
+    avifRWDataRealloc(profile, profileLength);
+    sprintf((char *)profile->data, "\n%s\n%08lu\n", profileName, (unsigned long)bytes->size);
```

Comment:
snprintf would likely be preferred here and line 401; sprintf will produce warnings in some configurations.